Experimental Separator Component

natalieethell opened the pull request:
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

This PR adds the experimental separator component. There are two main variations: vertical and horizontal. The default is horizontal, and the vertical variation can be obtained by passing in the `vertical` prop. A developer can also specify how they would like to align the text, using the `alignText` prop. Below I've included screenshots of the demo page in experiments.

![image](https://user-images.githubusercontent.com/14134000/46043827-ddfd6500-c0cd-11e8-85cf-8ca61b97a7dd.png)
![image](https://user-images.githubusercontent.com/14134000/46044058-93c8b380-c0ce-11e8-9c3a-482e2dffcd1c.png)


Some thing(s) to note:
* The height of the vertical separator is currently set to 200px. I'm not sure if we'd want some sort of default value here or if we'd like it to inherit the parent's height, so let me know if you have any thoughts on that!
* The background color of the text is set to white, however, I believe that we'd like this to always match the background color of the web app the separator is placed in. Would we want this to be a prop? Or is a white default okay?
* I set the color of the line to `theme.palette.neutralLight`. Let me know if we'd like to use a different color from palette instead!


###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/6471)

KevinTCoughlin commented on packages/experiments/src/components/Separator/Separator.types.ts:
```diff
@@ -0,0 +1,78 @@
+  /**
+   * How to align text in the separator.
+   */
+  alignText?: 'start' | 'center' | 'end';
```
👍

KevinTCoughlin commented on packages/experiments/src/components/Separator/Separator.types.ts:
```diff
@@ -0,0 +1,78 @@
+
+export interface ISeparatorProps extends React.Props<SeparatorBase> {
+  /**
+   * Optional callback to access the IChiclet interface. Use this instead of ref for accessing
```
nit: `IChiclet` => `ISeparator`

KevinTCoughlin commented on packages/experiments/src/components/Separator/Separator.styles.ts:
```diff
@@ -0,0 +1,83 @@
+        display: 'inline-block',
+        fontSize: '16px',
+        padding: '0 20px',
+        background: 'white'
```
Would `white` look strange if this were used on say a "dark theme"?

KevinTCoughlin commented on packages/experiments/src/components/Separator/Separator.types.ts:
```diff
@@ -0,0 +1,78 @@
+  className?: string;
+}
+
+export interface ISeparatorStyleProps {
```
You may consider using `Pick<T, K extends keyof T>` here where `T` is `ISeparatorProps` and the keys are common fields if it helps you.

I just remember it from our Styling Wiki: https://github.com/OfficeDev/office-ui-fabric-react/wiki/Component-Styling#icomponentstyleprops

KevinTCoughlin commented on packages/experiments/src/components/Separator/Separator.types.ts:
```diff
@@ -0,0 +1,78 @@
+  vertical?: boolean;
+}
+
+export interface ISeparatorStyles {
```
IIRC we recommend all `IComponentStyles` be `required` properties.

https://github.com/OfficeDev/office-ui-fabric-react/wiki/Component-Styling#icomponentstyles

natalieethell commented:
Changed the vertical separator to fill its container's height by default, fixed the background color of text to work with themes by changing it to `theme.semanticColors.bodyBackground`, changed ISeparatorStyleProps to use Pick, and made all ISeparatorStyles required.
